dsc_lib/src/parser/expressions.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use tree_sitter::Node;
+
+use crate::dscerror::DscError;
+use crate::functions::FunctionDispatcher;
+use crate::parser::functions::{Function, FunctionResult};
+
+#[derive(Clone)]
+pub struct Expression {
+    function: Function,
+    member_access: Option<Vec<String>>,
+}
+
+impl Expression {
+    /// Create a new `Expression` instance.
+    ///
+    /// # Arguments
+    ///
+    /// * `function_dispatcher` - The function dispatcher to use.
+    /// * `statement` - The statement that the expression is part of.
+    /// * `expression` - The expression node.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if the expression node is not valid.
+    pub fn new(function_dispatcher: &FunctionDispatcher, statement_bytes: &[u8], expression: &Node) -> Result<Self, DscError> {
+        let Some(function) = expression.child_by_field_name("function") else {
+            return Err(DscError::Parser("Function node not found".to_string()));
+        };
+        let function = Function::new(function_dispatcher, statement_bytes, &function)?;
+        let member_access = if let Some(member_access) = expression.child_by_field_name("members") {
```

Comment:
For better readability, please use different variable names here.
```suggestion
        let member_access = if let Some(members) = expression.child_by_field_name("members") {
```